tmt/steps/provision/bootc.py

```diff
@@ -0,0 +1,157 @@
+
+    add_deps: bool = field(
+        default=True,
+        is_flag=True,
+        option=('--add-deps'),
+        help="""
+             Add tmt dependencies to the supplied container image or image built
+             from the supplied Containerfile.
+             This will cause a derived image to be built from the supplied image.
+             """)
+
+    def set_image(self, image: str) -> None:
+        self.image = image
+
+
+@tmt.steps.provides_method('bootc')
+class ProvisionBootc(tmt.steps.provision.ProvisionPlugin[BootcData]):
+    """
+    Provision guest using bootc
+
+    Minimal configuration using the latest nothing image:
+
+    .. code-block:: yaml
+
+        provision:
+            how: bootc
+            containerfile: file://Containerfile
+
+    """
+
+    _data_class = BootcData
+    _guest_class = GuestTestcloud
+    _guest = None
+
+    # build a "modified" container image from the base image with tmt dependencies added
+    def _build_derived_image(self, base_image: str) -> str:
+        self._logger.debug("Building modified container image with necessary tmt packages/config")
+        containerfile_template = Template('''
+            FROM $base_image
+
+            RUN \
+            dnf -y install cloud-init rsync && \
+            ln -s ../cloud-init.target /usr/lib/systemd/system/default.target.wants && \
+            rm /usr/local -rf && ln -sr /var/usrlocal /usr/local && mkdir -p /var/usrlocal/bin
+        ''')
+        containerfile = containerfile_template.substitute({"base_image": base_image})
+        with open(f'{self.workdir}/Containerfile', 'w') as file:
+            file.write(containerfile)
+
+        image_tag = "localhost/tmtmodified"  # TODO: unique ID
+        tmt.utils.Command(
+            "podman", "build", f'{self.workdir}',
+            "-f", f'{self.workdir}/Containerfile',
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+
+        return image_tag
+
+    # build the "base" or user supplied container image
+    def _build_base_image(self, containerfile: str, workdir: str) -> str:
+        image_tag = "localhost/tmtbase"  # TODO: how do I get a unique ID here?
+        self._logger.debug("Building container image")
+        tmt.utils.Command(
+            "podman", "build", workdir,
+            "-f", containerfile,
+            "-t", image_tag
+            ).run(cwd=self.workdir, stream_output=True, logger=self._logger)
+        return image_tag
+
+    # build the bootc disk
+    def _build_bootc_disk(self, containerimage: str) -> None:
+        self._logger.debug("Building bootc disk image")
+        tmt.utils.Command(
+            "podman", "run", "--rm", "--privileged",
+            "-v", "/var/lib/containers/storage:/var/lib/containers/storage",
+            "--security-opt", "label=type:unconfined_t",
+            "-v", f"{self.workdir}:/output",
+            "quay.io/centos-bootc/bootc-image-builder:latest", "build",
```

Comment:
I've been trying to avoid people hardcoding this because we already have the situation where there are two builders (one upstream, one for RHEL). Exact same comment as https://github.com/containers/podman-bootc/pull/58#discussion_r1668661289

It's actually somewhat likely that we move out of the quay.io/centos-bootc namespace at some point; while we'd probably maintain a redirect for "a while", with the "consume a label" approach it automatically derives from the base image which avoids hardcoding entirely.

Reply:
Would it make sense to make the namespace configurable? In any case, having a default as a module-level constant should happen, but is it something the user might wish to override? E.g. `namespace: quay.io/acme/bootc`? I think it would be good to open this piece to users.

Reply:
added a parameter for the user to configure this in the latest push